NitroxServer/GameLogic/Bases/BaseManager.cs

```diff
@@ -33,29 +35,83 @@ public List<BasePiece> GetPartiallyConstructedPieces()
                 return new List<BasePiece>(partiallyConstructedPiecesById.Values);
             }
         }
-
         public void AddBasePiece(BasePiece basePiece)
         {
-            lock (partiallyConstructedPiecesById)
+            //anything that is not a window, hatch, or reinforcement get run normally
+            if (basePiece.TechType.Name != "BaseReinforcement" | basePiece.TechType.Name != "BaseHatch" | basePiece.TechType.Name != "BaseWindow")
+            {
+                //just in case some how a base piece is spawned at max completion
+                //the base piece will get added to at least one directory
+                if(basePiece.ConstructionAmount < 1.0f)
+                {
+                    lock (partiallyConstructedPiecesById)
+                    {
+                        partiallyConstructedPiecesById.Add(basePiece.Id, basePiece);
+                    }
+                }
+                else
+                {
+                    lock (partiallyConstructedPiecesById)
+                    {
+                        completedBasePieceHistory.Add(basePiece);
+                    }
+                }
+            }
+            //when adding base peices this tries to eliminate partially constructed parts that were once fully constructed
+            if (basePiece.ConstructionAmount > 0.4f && basePiece.ConstructionAmount != 1.0f)
+            {
+                basePiece.ConstructionAmount = 1.0f;
+                basePiece.ConstructionCompleted = true;
+            }
+            //I am not sure if this helps at all, but it is supposed to ensure that the windows, hatches, and reinforcements don't go missing
+            //and that they they are built to 100%
+            if (basePiece.TechType.Name == "BaseWindow" && completedBasePieceHistory.Contains(basePiece))
+            {
+                basePiece.ConstructionCompleted = false;
+                partiallyConstructedPiecesById.Add(basePiece.Id, basePiece);
+                basePiece.ConstructionAmount = 1.0f;
+                basePiece.ConstructionCompleted = true;
```

Comment:
I'm not sure what the purpose was to see it to false when subsequently overriding it to true.

Reply:
me neither...